cli/agent.go

```diff
@@ -36,7 +40,20 @@ func (c *AgentCommand) Execute(_ []string) error {
 	grip.SetThreshold(level.Debug)
 	grip.SetName("evg-agent")
 
-	// all we need is the host id and host secret
+	if c.NewAgent {
+		initialOptions := proto.Options{
+			APIURL:     c.ServiceURL,
+			HostID:     c.HostID,
+			HostSecret: c.HostSecret,
+			StatusPort: c.StatusPort,
+			LogPrefix:  c.LogPrefix,
+		}
+
+		agt := proto.New(initialOptions, client.NewCommunicator(c.ServiceURL))
```

Comment:
seems weird that we need to pass the service URL to the communicator and the agent class itself.